ext/openssl/openssl.c

```diff
@@ -1683,19 +1800,25 @@ PHP_FUNCTION(openssl_spki_export_challenge)
 /* {{{ Exports a CERT to file or a var */
 PHP_FUNCTION(openssl_x509_export)
 {
-	X509 * cert;
-	zval * zcert, *zout;
+	X509 *cert;
+	zend_object *cert_obj;
+	zend_string *cert_str;
+	zval *zout;
 	zend_bool notext = 1;
 	BIO * bio_out;
 
-	if (zend_parse_parameters(ZEND_NUM_ARGS(), "zz|b", &zcert, &zout, &notext) == FAILURE) {
-		RETURN_THROWS();
-	}
+	ZEND_PARSE_PARAMETERS_START(2, 3)
+		Z_PARAM_STR_OR_OBJ_OF_CLASS(cert_str, cert_obj, php_openssl_certificate_ce)
+		Z_PARAM_ZVAL(zout)
+		Z_PARAM_OPTIONAL
+		Z_PARAM_BOOL(notext)
+	ZEND_PARSE_PARAMETERS_END();
+
 	RETVAL_FALSE;
 
-	cert = php_openssl_x509_from_zval(zcert, 0, NULL);
+	cert = php_openssl_x509_from_param(cert_obj, cert_str);
 	if (cert == NULL) {
-		php_error_docref(NULL, E_WARNING, "Cannot get cert from parameter 1");
+		php_error_docref(NULL, E_WARNING, "X.509 Certificate cannot be retrieved");
```

Comment:
Should I change these kind of error messages to `argument #1 ($x509) must be a valid X.509 Certificate`?